### PR TITLE
Document key sharing events

### LINKED ISSUE #1183: Document key-share requests
Documentation: https://docs.google.com/document/d/1m4gQkcnJkxNuBmb5NoFCIadIY-DyqqNAS3lloE73BlQ
PRs: #1465

We should document the events used by riot to do keyshare requests

### PR DESCRIPTION
**Pull request by Zil0**
fix #1183 

Signed-off-by: Valentin Deniaud \<valentin.deniaud@inpt.fr\>

### CODE REVIEW

**Comment by KitsuneRal** [comment]
I'm not sure if we can use YAML instead of JSON in `event-schemas`. So far I've seen event schemas using only JSON (and being compliant with JSON Schema). YAML is an OpenAPI thing. While I as an author of an event schemas parser am perfectly fine with YAML for event schemas, other parsers based on JSON Schema will be hugely unimpressed.

**Comment by Zil0** [comment]
@KitsuneRal not really sure what you mean. From what I can tell a lot (most?) of the events defined under `event-schemas/schema/` are yaml files.

**Comment by KitsuneRal** [comment]
Ehm, for a good reason. Discard my blabber, I've been out of my mind apparently.

**Comment by uhoreg** [comment]
@Zil0, now that the main E2E PR is merged (:clap:), can you rebase this one and fix the conflicts?  Thanks.

**Comment by Zil0** [comment]
@uhoreg should be good :)

**Comment by turt2live** [comment]
(changelog would be awesome again :)

File: event-schemas/schema/m.forwarded_room_key [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      session_key:
+        type: string
+        description: The key to be exchanged.
+      sender_claimed_ed25519_key:
+        type: string
+        description: |-
+          The Ed25519 key of the device which initiated the session originally.
+          It is 'claimed' because the receiving device has no way to tell that the
+          original room_key actually came from a device which owns the private part of
+          this key unless they have done device verification.
```
Comment (uhoreg) [inline event-schemas/schema/m.forwarded_room_key]: I don't see how device verification helps with this (as I just commented in the Google Doc, and I guess is a question for @richvdh after the weekend)

File: event-schemas/schema/m.room_key_request [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          room_id:
+            type: string
+            description: The room where the key is used.
+          sender_key:
+            type: string
+            description: |-
+              The Curve25519 key of the device which initiated the session originally.
+          session_id:
+            type: string
+            description: The ID of the session holding the key.
```
Comment (uhoreg) [inline event-schemas/schema/m.room_key_request]: Same as my other comment above.

File: event-schemas/schema/m.forwarded_room_key [hunk 3]
```diff
@@ -0,0 +1,58 @@
+---
+allOf:
+  - $ref: core-event-schema/event.yaml
+
+description: |-
+  This event type is used to forward keys for end-to-end encryption. Typically
+  it is encrypted as an ``m.room.encrypted`` event.
+properties:
+  content:
+    properties:
```
[the remark this answers is not in the document]
Comment (Zil0) [reply event-schemas/schema/m.forwarded_room_key]: It is useless, see https://github.com/matrix-org/matrix-doc/issues/1249. I'm assuming the export format also includes the `chain_index`, haven't actually checked.

**Comment by Zil0** [comment]
fixed

**Pull request merged**